### PR TITLE
Add perf counter

### PR DESCRIPTION
**Pull request by xiaoling-yi**
This PR adds the performance counter for profiling. The performance counter is written into the specific CSR after the streamer is idle. And the performance counter written has higher priority than the host CSR request.

### CODE REVIEW

File: src/main/scala/streamer/StreamerTop.scala [hunk 1]
```diff
@@ -49,8 +49,9 @@ class StreamerTop(
 
   override val desiredName = params.tagName + "StreamerTop"
 
+  // add extra performance counter
   val csrNum: Int =
-    params.temporalDim + params.dataMoverNum * params.temporalDim + params.spatialDim.sum + params.dataMoverNum + 1
+    params.temporalDim + params.dataMoverNum * params.temporalDim + params.spatialDim.sum + params.dataMoverNum + 1 + 1
```
Comment (rgantonio) [inline src/main/scala/streamer/StreamerTop.scala:54]: +1 +1?? not +2? haha

File: src/main/scala/streamer/StreamerTop.scala [hunk 2]
```diff
@@ -67,7 +68,39 @@ class StreamerTop(
...
+
+  streamerIdle2Busy := streamer.io.busy_o && !RegNext(streamer.io.busy_o)
+  streamerBusy2Idle := !streamer.io.busy_o && RegNext(streamer.io.busy_o)
+
+  val performance_counter = RegInit(0.U(32.W))
+  when(streamer.io.busy_o) {
+    performance_counter := performance_counter + 1.U
+  }.elsewhen(streamerBusy2Idle) {
+    performance_counter := 0.U
+  }
```
Comment (rgantonio) [inline src/main/scala/streamer/StreamerTop.scala:88]: So this is to count how much time it took to process? Not how many cycles were successful yes?

File: src/main/scala/streamer/StreamerTop.scala [hunk 3]
```diff
@@ -67,7 +68,39 @@ class StreamerTop(
...
+
+  csr_config_in_req_valid := io.csr.req.valid
+  when(streamerBusy2Idle) {
+    csr_config_in_req_bits.addr := csrNum.U - 2.U
+    csr_config_in_req_bits.data := performance_counter
+    csr_config_in_req_bits.write := true.B
+  }.otherwise {
+    csr_config_in_req_bits := io.csr.req.bits
+  }
```
Comment (rgantonio) [inline src/main/scala/streamer/StreamerTop.scala:97]: Wait so when it's busy the addr and data keep requesting these data?

Comment (rgantonio) [reply src/main/scala/streamer/StreamerTop.scala:97]: Ah nvm so it's like the last state yes?

**Review by rgantonio — APPROVED** [review]
You need this fix??

**Comment by xiaoling-yi** [comment]
Discussed offline!

**Pull request merged**